sdk/cosmos/azure-cosmos-encryption/src/test/java/com/azure/cosmos/implementation/encryption/EncryptionTests.java

```diff
@@ -522,6 +590,58 @@ public void createItemEncrypt_readItemDecrypt() throws Exception {
         assertThat(readWithoutDecryption.sensitive).isNull();
     }
 
+    // async write, upsert, read
+    @Test(groups = { "encryption" }, timeOut = TIMEOUT)
+    public void asyncCreateItemEncrypt_readItemDecrypt() throws Exception {
+        EncryptionItemRequestOptions requestOptions = new EncryptionItemRequestOptions();
+        EncryptionOptions encryptionOptions = new EncryptionOptions();
+        encryptionOptions.setPathsToEncrypt(ImmutableList.of("/Sensitive"));
+
+        encryptionOptions.setDataEncryptionKeyId(dekId);
+        encryptionOptions.setEncryptionAlgorithm(CosmosEncryptionAlgorithm.AEAES_256_CBC_HMAC_SHA_256_RANDOMIZED);
+        requestOptions.setEncryptionOptions(encryptionOptions);
+
+        TestDoc properties = getItem(UUID.randomUUID().toString());
```

Comment:
Instead of using blocking calls, since now we are going asynchronous, please try using `StepVerifier` from reactor-test. 
Its actually built in for testing async APIs. We have used it in azure-spring-data-cosmos-test and also in azure-cosmos. 
Please take a look at `VeryLargeDocumentQueryTest.java` and any reactive test in azure-spring-data-cosmos-test - `ReactiveCosmosTemplateIT`, `ReactiveCourseRepositoryIT`, etc.

Reply:
good suggestion. Moved to `StepVerifier`.